SPEC.rdoc

```diff
@@ -135,80 +197,71 @@ The special response header +rack.hijack+ must only be set if the request +env+
 
 == The Response
 
+Outgoing HTTP responses are generated from the response tuple generated by the application. The response tuple is an +Array+ of three elements, which are: the HTTP status, the headers, and the response body. The Rack application is responsible for ensuring that the response tuple is well-formed and should follow the rules set out in the Rack specification.
+
 === The Status
 
 This is an HTTP status. It must be an Integer greater than or equal to 100.
 
 === The Headers
 
-The headers must be a unfrozen Hash.
-The header keys must be Strings.
-Special headers starting "rack." are for communicating with the
-server, and must not be sent back to the client.
-The header must not contain a +Status+ key.
-Header keys must conform to RFC7230 token specification, i.e. cannot
-contain non-printable ASCII, DQUOTE or "(),/:;<=>?@[\]{}".
-Header keys must not contain uppercase ASCII characters (A-Z).
-Header values must be either a String instance,
-or an Array of String instances,
-such that each String instance must not contain characters below 037.
+The headers must be a unfrozen +Hash+. The header keys must be +String+ objects. Special headers starting <tt>rack.</tt> are for communicating with the server, and must not be sent back to the client.
+
+* The headers must not contain a <tt>"status"</tt> key.
+* Header keys must conform to RFC7230 token specification, i.e. cannot contain non-printable ASCII, DQUOTE or <tt>(),/:;<=>?@[\]{}</tt>.
+* Header keys must not contain uppercase ASCII characters (A-Z).
+* Header values must be either a +String+ instance, or an +Array+ of +String+ values, such that each String instance must not contain characters below 037.
 
 ==== The +content-type+ Header
 
-There must not be a <tt>content-type</tt> header key when the +Status+ is 1xx, 204, or 304.
+There must not be a <tt>content-type</tt> header key when the status is 1xx, 204, or 304.
 
 ==== The +content-length+ Header
 
-There must not be a <tt>content-length</tt> header key when the +Status+ is 1xx, 204, or 304.
+There must not be a <tt>content-length</tt> header key when the status is 1xx, 204, or 304.
 
 ==== The +rack.protocol+ Header
 
-If the +rack.protocol+ header is present, it must be a +String+, and must be one of the values from the +rack.protocol+ array from the environment.
+If the <tt>rack.protocol</tt> header is present, it must be a +String+, and must be one of the values from the +rack.protocol+ array from the environment.
 
 Setting this value informs the server that it should perform a connection upgrade. In HTTP/1, this is done using the +upgrade+ header. In HTTP/2, this is done by accepting the request.
 
 === The Body
 
-The Body is typically an +Array+ of +String+ instances, an enumerable that yields +String+ instances, a +Proc+ instance, or a File-like object.
+The Body is typically an +Array+ of +String+ values, an enumerable that yields +String+ values, a +Proc+ instance, or a File-like object.
```

Comment:
I think `instances` is better than `values` here, so I would revert this change.

Reply:
I'd just like to be consistent, as I think whatever we do here, as long as it's consistent.

Reply:
It's still like this to change back to `instances`.